src/ucp/wireup/select.c

```diff
@@ -1000,6 +1060,15 @@ ucs_status_t ucp_wireup_select_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
                                                           address_list,
                                                           lane_descs,
                                                           key->num_lanes);
+    /* Cut slowest rndv lanes by configuration value */
+    for (lane = ep->worker->context->config.ext.max_rndv_lanes; lane < UCP_MAX_RAILS; lane++) {
+        key->rndv_lanes[lane] = UCP_NULL_LANE;
+    }
+
+    for (key->num_rndv_lanes = 0;
```

Comment:
seems like if you would really "cut" the array and remove some elements from key->rndv_lanes[], they would still remain in key->lanes[], which would potentially lead to creating un-needed connections. Need to remove the low-score rndv lanes before.

Reply:
no more relevant - wireup updated